common/dev/adm1272.c

```diff
@@ -256,6 +259,95 @@ int adm1272_read_iin(sensor_cfg *cfg, float *val)
 	return 0;
 }
 
+static int adm1272_read_ein(sensor_cfg *cfg, float *val)
+{
+	CHECK_NULL_ARG_WITH_RETURN(val, -1);
+	CHECK_NULL_ARG_WITH_RETURN(cfg, -1);
+	CHECK_NULL_ARG_WITH_RETURN(cfg->init_args, -1);
+
+	adm1272_init_arg *init_arg = (adm1272_init_arg *)cfg->init_args;
+
+	uint8_t retry = 5;
+	int ret = 0;
+	I2C_MSG msg = { 0 };
+
+	uint32_t energy = 0, rollover = 0, sample = 0;
+	uint32_t pre_energy = 0, pre_rollover = 0, pre_sample = 0;
+	uint32_t sample_diff = 0;
+	double energy_diff = 0;
+
+	/* Read EIN */
+	msg.bus = cfg->port;
+	msg.target_addr = cfg->target_addr;
+	msg.tx_len = 1;
+	msg.rx_len = 7;
+	msg.data[0] = cfg->offset;
+
+	ret = i2c_master_read(&msg, retry);
+	if (ret != 0) {
+		LOG_ERR("Read EIN fail, ret: %d, sensor num: 0x%x", ret, cfg->num);
+		return -1;
+	}
+
+	/* Record the previous and current data */
+#ifdef MORE_THAN_ONE_ADM1272
+	pre_energy = init_arg->last_energy;
+	pre_rollover = init_arg->last_rollover;
+	pre_sample = init_arg->last_sample;
+
+	init_arg->last_energy = energy = (msg.data[2] << 8) | msg.data[1];
+	init_arg->last_rollover = rollover = msg.data[3];
+	init_arg->last_sample = sample = (msg.data[6] << 16) | (msg.data[5] << 8) | msg.data[4];
+
+	if (init_arg->is_record_ein != true) {
+		init_arg->is_record_ein = true;
+		return -1;
+	}
+#else
+	static bool is_record_ein = false;
+	static uint32_t last_energy = 0, last_rollover = 0, last_sample = 0;
+
+	pre_energy = last_energy;
+	pre_rollover = last_rollover;
+	pre_sample = last_sample;
+
+	last_energy = energy = (msg.data[2] << 8) | msg.data[1];
+	last_rollover = rollover = msg.data[3];
+	last_sample = sample = (msg.data[6] << 16) | (msg.data[5] << 8) | msg.data[4];
+
+	if (is_record_ein == false) {
+		is_record_ein = true;
+		return -1;
```

Comment:
Can you please define some pass and fail macros and use them in caller and callee at all places. This way code readability doesn't look good.

Reply:
Hi @jagpalgill ,
Modify with your suggestions, please help to review this commit again, thank you.

Note：
Review dog report：Checking if unsigned expression 'rollover/sample/energy' is less than zero.
But rollover/sample/energy is an unsigned integer, it will be greater than or equal to zero.